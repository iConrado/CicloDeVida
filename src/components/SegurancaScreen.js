import React from 'react';
import { ScrollView, View, Text, TextInput, Image } from 'react-native';

import styles from './functions/styles';
import Cabecalho from './functions/Cabecalho';
import Rodape from './functions/Rodape';
import Carregando from './functions/Carregando';
import EstiloVoltar from './functions/EstiloVoltar';
import ModalErro from './functions/ModalErro';
import controle from './functions/controle';
import Ciclo from './functions/Ciclo';
import monetizar from './functions/monetizar';
import desmonetizar from './functions/desmonetizar';

const imgConvenio = require('../imgs/ic_vallet_white.png');
const imgVida = require('../imgs/ic_people_white.png');
const imgImoveis = require('../imgs/ic_home_white.png');
const imgAuto = require('../imgs/ic_car_white.png');

const C = new Ciclo();
let objErro = {};
const tmpComprometimento = [];

export default class SegurancaScreen extends React.Component {
  static navigationOptions = {
    //eslint-disable-line
    headerTitle: Cabecalho('Consultoria Ciclo de Vida'),
    headerBackTitle: 'Voltar',
    headerTintColor: EstiloVoltar.hTintColor,
    headerStyle: EstiloVoltar.hStyle,
  };

  constructor(props) {
    super(props);
    this.state = {
      carregado: false,
      modalErro: false,
      saude: 0,
      comprometimento: 0,
    };

    this.fechaErro = this.fechaErro.bind(this);
    this.abreErro = this.abreErro.bind(this);
    this.proxTela = this.proxTela.bind(this);
  }

  componentDidMount() {
    this.montagem();
  }

  montagem() {
    tmpComprometimento[0] = this.state.saude;
    this.comprometimentoAtual();
    this.setState({ carregado: true });
  }

  abreErro(e, tipo) {
    objErro = e;
    this.setState({ modalErro: true });
    switch (tipo) {
      case 0:
        console.log('Retorno 0');
        break;
      default:
        console.log('Retorno default');
    }
  }

  fechaErro() {
    this.setState({ modalErro: false });
    objErro = {};
  }

  sugestLim() {
    return C.sugestaoLimSeg();
  }

  calculaVida() {
    return C.seguroVida();
  }

  calculaImoveis() {
    return C.seguroImoveis();
  }

  calculaAuto() {
    return C.seguroAuto();
  }

  calculaComprTotal() {
    const convenio = this.state.saude;
    const vida = this.calculaVida();
    const imoveis = this.calculaImoveis();
    const auto = this.calculaAuto();
    const soma = convenio + vida + imoveis + auto;
    const compr = C.comprometimentoGasto(soma);

    return parseFloat(compr * 100, 10)
      .toFixed(1)
      .replace('.', ',');
  }

  calculaPatrimProt() {
    const conv = this.state.saude;
    return C.patrimonioProt(conv);
  }

  comprometimento(valor) {
    const compr = C.comprometimentoGasto(valor);

    return compr;
  }

  calculaCompromSaude() {
    tmpComprometimento[0] = this.state.saude;
    this.comprometimentoAtual();
  }

  comprometimentoAtual() {
    let valor = 0;
    if (tmpComprometimento[0] !== undefined) {
      valor = tmpComprometimento.reduce((prevVal, elem) => prevVal + elem);
    }
    const compr = C.comprometimentoAtual('Seguranca', valor);
    this.setState({ comprometimento: compr });
  }

  proxTela(tela) {
    // Função que valida os campos e submete os dados para registro na classe de negócio.
    // Em caso de algum retorno com erro, executa a abertura da tela de erros.
    const { navigate } = this.props.navigation;

    const saude = this.state.saude;

    // Validação das regras de negócio, registro e gravação de log
    if (!controle(this.abreErro, C, C.setSaude, saude)) {
      return false;
    }

    navigate(tela);
  }

  render() {
    if (!this.state.carregado) {
      return <Carregando />;
    }
    return (
      <View style={styles.tela}>
        <ScrollView style={styles.scroll} contentContainerStyle={styles.container}>
          {/* Camada Modal que intercepta erros e exibe uma mensagem personalizada na tela */}
          <ModalErro visivel={this.state.modalErro} fechar={this.fechaErro} objErro={objErro} />
          {/* **************************************************************************** */}

          <View style={styles.viewTitulo}>
            <Text style={styles.titulo}>Segurança</Text>
          </View>

          <View style={styles.viewHorizontal}>
            <View style={styles.viewCompHoriz}>
              <Text style={[styles.segur_lbCalculo, styles.label]}>Sugestão de limite mensal:</Text>
            </View>
            <View style={styles.viewCompHoriz}>
              <Text style={styles.segur_txDireita}>{`${this.comprometimento(this.sugestLim()) * 100}%`}</Text>
              <Text style={styles.segur_txDireita}>{monetizar(this.sugestLim())}</Text>
            </View>
          </View>

          <View style={styles.espacador} />
          <View style={styles.separador} />

          <View style={styles.viewHorizontal}>
            <View style={styles.viewIcone}>
              <Image style={styles.imgIcone} source={imgConvenio} />
            </View>
            <View style={styles.viewPosIcone}>
              <View style={styles.viewVertical}>
                <Text style={styles.label}>Convênios de saúde:</Text>
                <TextInput
                  ref="saude"
                  style={styles.segur_input}
                  keyboardType="numeric"
                  maxLength={10}
                  autoCorrect={false}
                  selectTextOnFocus
                  underlineColorAndroid="#EAEAEA"
                  onChangeText={text => this.setState({ saude: desmonetizar(text) })}
                  onEndEditing={() => this.calculaCompromSaude()}
                  value={monetizar(this.state.saude)}
                />
              </View>
            </View>
          </View>
          <View style={styles.separador} />
          <View style={styles.espacador} />

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.segur_viewEsquerda} />
              <View style={styles.viewPosIcone}>
                <View style={styles.viewHorizontal}>
                  <Text style={[styles.label, styles.negrito, styles.segur_lbSeguridade]}>Seguros</Text>
                  <Text style={[styles.label, styles.negrito, styles.segur_txValor]}>Valor mensal</Text>
                  <Text style={[styles.label, styles.negrito, styles.segur_txValor]}>Patrimonio protegido</Text>
                </View>
              </View>
            </View>
          </View>

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.viewIcone}>
                <Image style={styles.imgIcone} source={imgVida} />
              </View>
              <View style={styles.viewPosIcone}>
                <View style={styles.viewHorizontal}>
                  <Text style={[styles.label, styles.segur_lbSeguridade]}>Vida (24x renda):</Text>
                  <Text style={styles.segur_txValor}>{monetizar(this.calculaVida())}</Text>
                  <Text style={styles.segur_txValor}>{monetizar(C.getSalLiq() * 24)}</Text>
                </View>
              </View>
            </View>
          </View>

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.viewIcone}>
                <Image style={styles.imgIcone} source={imgImoveis} />
              </View>
              <View style={styles.viewPosIcone}>
                <View style={styles.viewHorizontal}>
                  <Text style={[styles.label, styles.segur_lbSeguridade]}>Residencial:</Text>
                  <Text style={styles.segur_txValor}>{monetizar(this.calculaImoveis())}</Text>
                  <Text style={styles.segur_txValor}>{monetizar(C.getImoveis())}</Text>
                </View>
              </View>
            </View>
          </View>

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.viewIcone}>
                <Image style={styles.imgIcone} source={imgAuto} />
              </View>
              <View style={styles.viewPosIcone}>
                <View style={styles.viewHorizontal}>
                  <Text style={[styles.label, styles.segur_lbSeguridade]}>Auto:</Text>
                  <Text style={styles.segur_txValor}>{monetizar(this.calculaAuto())}</Text>
                  <Text style={styles.segur_txValor}>{monetizar(C.getVeiculos())}</Text>
                </View>
              </View>
            </View>
          </View>

          <View style={styles.espacador} />
          <View style={styles.separador} />

          <View style={styles.viewHorizontal}>
            <View style={styles.viewCompHoriz}>
              <Text style={[styles.segur_lbCalculo, styles.label, styles.segur_lbTotal]}>Comprometimento com segurança:</Text>
            </View>
            <View style={styles.viewCompHoriz}>
              <Text style={[styles.segur_txDireita, styles.segur_lbTotal]}>{`${this.calculaComprTotal()}%`}</Text>
            </View>
          </View>

          <View style={styles.viewHorizontal}>
            <View style={styles.viewCompHoriz}>
              <Text style={[styles.segur_lbCalculo, styles.label, styles.segur_lbTotal]}>Patrimônio protegido:</Text>
            </View>
            <View style={styles.viewCompHoriz}>
              <Text style={[styles.segur_txDireita, styles.segur_lbTotal]}>{monetizar(this.calculaPatrimProt())}</Text>
            </View>
          </View>

          <View style={styles.separador} />
        </ScrollView>

        <Rodape valor={this.state.comprometimento} funcProxTela={this.proxTela} tela="Consumo" />
      </View>
    );
  }
}
