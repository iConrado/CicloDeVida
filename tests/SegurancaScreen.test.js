import renderer from 'react-test-renderer';

import React from 'react';
import SegurancaScreen from '../src/components/SegurancaScreen';


it('renders without crashing', () => {
  const rendered = renderer.create(<SegurancaScreen />).toJSON();
  expect(rendered).toMatchSnapshot();
});
