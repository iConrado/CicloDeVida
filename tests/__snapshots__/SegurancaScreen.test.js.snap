// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingHorizontal": 10,
        "paddingTop": 5,
      }
    }
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000",
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 0.8,
                "position": "absolute",
                "width": "100%",
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 10,
                "justifyContent": "center",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
                "position": "absolute",
                "width": "80%",
                "zIndex": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                  "marginBottom": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#5C86F5",
                  "borderRadius": 2,
                  "elevation": 4,
                  "marginTop": 10,
                  "minWidth": "30%",
                  "opacity": 1,
                  "paddingVertical": 4,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginHorizontal": 5,
                  }
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </Modal>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 5,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#14567A",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Segurança
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "#666",
                },
              ]
            }
          >
            Sugestão de limite mensal:
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": "100%",
              }
            }
          >
            0%
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": "100%",
              }
            }
          >
            R$ 0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#48AEB5",
              "borderRadius": 13,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/imgs/ic_vallet_white.png",
              }
            }
            style={
              Object {
                "height": 35,
                "width": 35,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexShrink": 1,
              "justifyContent": "space-between",
              "paddingLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Convênios de saúde:
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              keyboardType="numeric"
              maxLength={10}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              selectTextOnFocus={true}
              style={
                Object {
                  "borderBottomColor": "#AAA",
                  "borderBottomWidth": 1,
                  "height": 30,
                  "paddingHorizontal": 5,
                  "width": "100%",
                }
              }
              underlineColorAndroid="#EAEAEA"
              value="R$ 0"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": 40,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "width": "40%",
                    },
                  ]
                }
              >
                Seguros
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "30%",
                    },
                  ]
                }
              >
                Valor mensal
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "30%",
                    },
                  ]
                }
              >
                Patrimonio protegido
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/imgs/ic_people_white.png",
                }
              }
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "width": "40%",
                    },
                  ]
                }
              >
                Vida (24x salário bruto):
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/imgs/ic_home_white.png",
                }
              }
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "width": "40%",
                    },
                  ]
                }
              >
                Residencial:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/imgs/ic_car_white.png",
                }
              }
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#666",
                    },
                    Object {
                      "width": "40%",
                    },
                  ]
                }
              >
                Auto:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "30%",
                  }
                }
              >
                R$ 0
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "#666",
                },
                Object {
                  "fontWeight": "bold",
                  "height": 35,
                },
              ]
            }
          >
            Comprometimento com segurança:
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                },
                Object {
                  "fontWeight": "bold",
                  "height": 35,
                },
              ]
            }
          >
            0,0%
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "#666",
                },
                Object {
                  "fontWeight": "bold",
                  "height": 35,
                },
              ]
            }
          >
            Patrimônio protegido:
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                },
                Object {
                  "fontWeight": "bold",
                  "height": 35,
                },
              ]
            }
          >
            R$ 0
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#D0D0D0",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "height": 60,
        "marginTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 6,
          "flexDirection": "row",
          "paddingHorizontal": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 7,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Comprometimento de renda atual:
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          0
          %
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 4,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5C86F5",
            "borderRadius": 2,
            "elevation": 4,
            "height": 35,
            "justifyContent": "center",
            "maxWidth": "90%",
            "minWidth": "80%",
            "opacity": 1,
            "paddingVertical": 4,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 5,
            }
          }
        >
          PRÓXIMA ETAPA
        </Text>
      </View>
    </View>
  </View>
</View>
`;
